addons/mail/static/tests/gif_picker/gif_picker_test.js

```diff
@@ -0,0 +1,125 @@
+/* @odoo-module */
+
+import {
+    click,
+    insertText,
+    start,
+    startServer,
+    waitUntil,
+} from "@mail/../tests/helpers/test_utils";
+import { patchUiSize, SIZES } from "../helpers/patch_ui_size";
+
+QUnit.module("gif");
+
+QUnit.test("composer should display an emoji button", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "" });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    assert.containsOnce($, "button[aria-label='Gifs']");
+});
+
+QUnit.test("Composer gif button should open the gif picker", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "" });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click("button[aria-label='Gifs']");
+    assert.containsOnce($, ".o-discuss-gif-picker-popover");
+});
+
+QUnit.test("Searching for a gif", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "" });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click("button[aria-label='Gifs']");
+    await insertText("input[placeholder='Search for a gif']", "search");
+    assert.containsOnce($, ".o-discuss-gif-picker-popover .fa-arrow-left");
```

Comment:
Icon is not super meaningful. so it's unclear why this is functionally relevant to assert that.
If you relied on an `aria-label`/`title`, or another named test that clarifies what this arrow does, this would be a better test.